packages/web3-providers/src/helpers/captureAsyncTransaction.ts

```diff
@@ -0,0 +1,36 @@
+import { Flags } from '@masknet/flags'
+
+export async function captureAsyncTransaction<T>(name: string, toBeResolved: Promise<T>): Promise<T> {
+    if (!Flags.sentry_enabled) return toBeResolved
+    if (!Flags.sentry_async_transaction_enabled) return toBeResolved
+
+    const startAt = Date.now()
+
+    let hasError = false
+    let capturedError: Error | undefined
+
+    try {
+        const resolved = await toBeResolved
+        return resolved
+    } catch (error) {
+        if (error instanceof Error) {
+            capturedError = error
+        }
+        hasError = true
+        throw error
+    } finally {
+        const transaction = Sentry.startTransaction({
```

Comment:
sentry transaction is used to track the timeline of some events. if it is created and finished with only 1 event, you may not need it.